pipelines/zendesk/__init__.py

```diff
@@ -0,0 +1,253 @@
+    @:param zendesk_client: Zenpy type object, used to make calls to Zendesk API through zenpy module
+    @:param start_date: Datetime object that serves as the starting date of when to retrieve objects
+    @:returns: Generator returning dicts for every row of data
+    """
+    # grab the start time from last dlt load if not filled, if that is also empty then use the first day of the millennium as the start time instead
+    if not start_time:
+        # the dlt state will store a timestamp, we need to convert back to datetime for Zenpy
+        start_time = from_timestamp(dlt.state().setdefault("last_load_chats", FIRST_DAY_OF_MILLENNIUM.timestamp()))
+    all_chats = zendesk_client.chats.incremental(start_time=start_time)
+    dict_generator = basic_load(resource_api=all_chats)
+    yield from dict_generator
+    # update the last load time in the dlt state and save as timestamp since datetime is not supported by dlt state
+    dlt.state()["last_load_chats"] = now().timestamp()
+
+
+# Talk resources
+def talk_resource(zendesk_client: ZendeskAPIClient, talk_endpoint_name: str, talk_endpoint: str) -> Iterator[TDataItem]:
+    """
+    Called as a dlt resource. Generic loader for a ZendeskTalk endpoint.
+    @:param zendesk_client: ZendeskAPIClient object that makes api calls to Zendesk Talk API
+    @:param talk_endpoint_name: The name of the talk_endpoint
+    @:param talk_endpoint: The actual url ending of this endpoint
+    @:returns: Generator of dicts
+    """
+    my_req = zendesk_client.make_request(endpoint=talk_endpoint, data_point_name=talk_endpoint_name)
+    if my_req:
+        my_pages = [page for page in my_req]
+        for page in my_pages:
+            # multiple records in page
+            if isinstance(page, list):
+                for record in page:
+                    yield record
+            # only 1 record in the page, just yield page
+            else:
+                yield page
+
+
+def talk_incremental_resource(zendesk_client: ZendeskAPIClient, talk_endpoint_name: str, talk_endpoint: str, start_date: Optional[DateTime] = None) -> Iterator[TDataItem]:
+    """
+    Called as a dlt resource. Generic loader for a ZendeskTalk endpoint.
+    @:param zendesk_client: ZendeskAPIClient object that makes api calls to Zendesk Talk API
+    @:param talk_endpoint_name: The name of the talk_endpoint
+    @:param talk_endpoint: The actual url ending of this endpoint
+    @:returns: Generator of dicts
+    """
+    # grab the start time from last dlt load if not filled, if that is also empty then use the first day of the millennium as the start time instead
+    if not start_date:
+        final_start_date = dlt.state().setdefault(f"last_load_talk_{talk_endpoint_name}", FIRST_DAY_OF_MILLENNIUM.timestamp())
+    else:
+        final_start_date = start_date.timestamp()
+    my_req = zendesk_client.make_request_incremental(endpoint=talk_endpoint, data_point_name=talk_endpoint_name, start_date=final_start_date)
```

Comment:
can't you just call `talk_resource` instead of duplicating code? parametrize it if necessary